packages/core/core/src/assetUtils.js

```diff
@@ -194,3 +203,41 @@ export async function getConfig(
 
   return conf;
 }
+
+export function getInvalidationId(invalidation: RequestInvalidation): string {
+  switch (invalidation.type) {
+    case 'file':
+      return 'file:' + invalidation.filePath;
+    case 'env':
+      return 'env:' + invalidation.key;
+    default:
+      throw new Error('Unknown invalidation type: ' + invalidation.type);
+  }
+}
+
+export async function getInvalidationHash(
+  invalidations: Array<RequestInvalidation>,
+  options: ParcelOptions,
+): Promise<string> {
+  let sortedInvalidations = invalidations
+    .slice()
+    .sort((a, b) => (getInvalidationId(a) < getInvalidationId(b) ? -1 : 1));
+
+  let hash = crypto.createHash('md5');
+  for (let invalidation of sortedInvalidations) {
+    switch (invalidation.type) {
+      case 'file':
+        hash.update(
+          await md5FromFilePath(options.inputFS, invalidation.filePath),
```

Comment:
We could probably cache these hashes somewhere at least during a transformation run so we aren't recomputing them all the time if it becomes a perf problem...